lib/ecto/repo/model.ex

```diff
@@ -182,11 +181,32 @@ defmodule Ecto.Repo.Model do
     put_in model.__meta__.state, :loaded
   end
 
-  defp merge_into_changeset(struct, fields, changeset) do
-    changes = Map.take(struct, fields)
+  defp merge_into_changeset(struct, fields, embeds, changeset) do
+    changes =
+      struct
+      |> Map.take(fields)
+      |> Map.drop(embeds)
+
     update_in changeset.changes, &Map.merge(changes, &1)
   end
 
+  defp apply_embedded_callbacks(embeds, changeset, callback) do
```

Comment:
Let's add a clause that match on empty embeds `[]` and return the changeset straight away.